### PR TITLE
My Jetpack: compute Search plugin price based on prices tier

### PR DESCRIPTION
**Pull request by retrofox**
We are not considering the price tier when computing the price for the Search plugin. This PR takes it over.

Some considerations:
* I've copied [the class](https://github.com/Automattic/jetpack/blob/master/projects/packages/search/src/class-stats.php) to compute the record count from the Search package.
* The data to use to compute the price is based on what the `/sites/$site/products` returns on the response body
* The endpoint mentioned above does not contain `introductory_offer` but contains the `sale_coupon` data. We use this data (the same [approach that we use in the Search plugin](https://github.com/Automattic/jetpack/blob/17079078398c920c0bfc222b5020954ab0dbfd6a/projects/packages/search/src/class-product.php#L111-L123))

Fixes #

#### Changes proposed in this Pull Request:

* My Jetpack: compute Search plugin price based on prices tier

#### Testing instructions:

* Go to My jetpack overview
* Test with a site without the Search product
* Confirm that the price shown in the UI is consistent with the record_count

0 / 100 | 101 / 1000 | 1001 / 10000 | 10001 / 100000 | 100000 / 1000000
------|------|------|-------|-------
image | image | image | image | image

You can take a look at the data populated by the client.
* Open the dev console
* run `wp.data.select( 'my-jetpack' ).getProduct( 'search' ).pricingForUi`

image

The image below shows the data for the higher tier

### CODE REVIEW

File: projects/packages/my-jetpack/src/products/class-search-stats.php [hunk 1]
```diff
@@ -0,0 +1,89 @@
+<?php
+/**
+ * Get search stats for use in the wp-admin dashboard.
+ *
+ * @package automattic/jetpack-search
```
Comment (jsnmoon) [inline projects/packages/my-jetpack/src/products/class-search-stats.php:5]: Do we want to update this to `@package automattic/my-jetpack`?

**Review by jsnmoon — DISMISSED** [review]
Changes look reasonable and work as described. 

I suspect Search will want to deprecate [its product class](https://github.com/Automattic/jetpack/blob/master/projects/packages/search/src/class-product.php) and rely wholly on My Jetpack for product pricing information in the future?

File: projects/packages/my-jetpack/src/products/class-search-stats.php (hunk 1 above)
Comment (retrofox) [reply projects/packages/my-jetpack/src/products/class-search-stats.php:5]: yes, we do. Thanks

File: projects/packages/my-jetpack/src/products/class-search.php [hunk 2]
```diff
@@ -106,13 +105,52 @@ public static function get_features() {
...
+		array_multisort( array_column( $price_tier_list, 'maximum_units' ), SORT_ASC, $price_tier_list );
+
+		// Pick the first tier that is less than or equal to the record count.
+		foreach ( $product->price_tier_list as $price_tier ) {
+			if ( $record_count <= $price_tier->maximum_units ) {
+				break;
+			}
+		}
+
+		// Computhe the minimum price.
```
Comment (kangzj) [inline projects/packages/my-jetpack/src/products/class-search.php:137]:
```suggestion
		// Compute the minimum price.
```

**Comment by retrofox** [comment]
> I suspect Search will want to deprecate [its product class](https://github.com/Automattic/jetpack/blob/master/projects/packages/search/src/class-product.php) and rely wholly on My Jetpack for product pricing information in the future?

Yeah, that makes sense to me since AFAIK all Jetpack plugins will rely on My Jetpack.

**Review by kangzj — DISMISSED** [review]
It looks good to me, only a possible typo as commented.
Thanks for working on this @retrofox ❤️

**Review by kangzj — APPROVED** [review]
LGTM

**Review by kraftbj — APPROVED** [review]

**Comment by retrofox** [comment]
@leo, I'd like to know your opinion here about the approach. Feel free to merge it.

**Review by leogermani — APPROVED** [review]
It looks good to me.

I think we could reorganize the files a bit better later, the `products` folder is getting crowded.

I confirmed this will not introduce new requests to WPCOM, as we are caching the results of get_products.

So it's a good time to remember that we are caching these values for 7 days! So any change in prices might take a while to propagate, we could ask around to see if we should reduce this.

Nice job!

**Comment by retrofox** [comment]
> So it's a good time to remember that we are caching these values for 7 days! So any change in prices might take a while to propagate, we could ask around to see if we should reduce this.

Do you think whether is possible to find a way to imperatively clear the cache, for instance, when the price values change?

> Nice job!

Thanks

**Pull request merged**